project/XNATDataImport.py

```diff
@@ -0,0 +1,64 @@
+import os
+import logging
+import pandas as pd
+import numpy as np
+
+from utils.tools import DataBuilderXNAT
+from xnat.mixin import ImageScanData, SubjectData
+from typing import List, Dict
+
+logger = logging.getLogger(__name__)
+
+class XNATDataImport():
+    
+    def __init__(self, xnat_configuration: dict = None, num_workers: int = 4):
+        self.xnat_configuration = xnat_configuration
+        self.num_workers = num_workers
+
+    def import_xnat_data(self):
+        actions = [
+            (self.fetch_xr, "image"),
+            (self.fetch_label, "label"),
+        ]
+
+        data_builder = DataBuilderXNAT(
+            self.xnat_configuration, actions=actions, num_workers=self.num_workers
+        )
+
+        data_builder.fetch_data()
+        return(data_builder.dataset)
+
+    @staticmethod
+    def fetch_xr(subject_data: SubjectData = None) -> List[ImageScanData]:
```

Comment:
"CR" and "DX" are I assume specific for X-rays. I recommend we add various `fetch` functions with good documentation and prompt the developer of the specific project to choose and adapt the available functions as applicable for their project.

Reply:
Create as separate issue!